src/scripts/cacheProposalTxSelectors.ts

```diff
@@ -0,0 +1,581 @@
+import * as CoreContractAbis from '@celo/abis';
+import 'dotenv/config';
+import { eq } from 'drizzle-orm';
+import fs from 'fs';
+import path from 'path';
+import database from 'src/config/database';
+import { eventsTable } from 'src/db/schema';
+import { getProposalTransactions } from 'src/features/governance/utils/transactionDecoder';
+import { logger } from 'src/utils/logger';
+import { fileURLToPath } from 'url';
+import { Abi, keccak256, toHex } from 'viem';
+// @ts-ignore
+BigInt.prototype.toJSON = function () {
+  return this.toString();
+};
+
+if (typeof process.env.NEXT_PUBLIC_CELOSCAN_API_KEY === 'string') {
+  logger.info('Using celoscan with API key');
+} else {
+  logger.warn(
+    'Celoscan api key not loaded. Votes will not be updated. You probably need a .env file (not local.env). To pull secrets see https://vercel.com/docs/cli/env',
+  );
+}
+
+const __filename = fileURLToPath(import.meta.url); // get the resolved path to the file
+const __dirname = path.dirname(__filename); // get the name of the directory
+
+const SELECTORS_OUT_PATH = path.resolve(__dirname, '../config/selectors.json');
+
+async function main() {
```

Comment:
when does this run? locally  or in CI? can you add some details about how to run and when ?